nougat/evaluete.py

```diff
@@ -349,3 +349,77 @@ def computeGC(sequence):
     gcFraction = float(gcCount) / totalBaseCount
     return gcFraction
 
+
+def computeAssemblyStats(sample_config):
+
+    outfile = os.path.join("contig_stats", "contiguity.out")
+    if not os.path.exists("contig_stats"):
+        os.makedirs("contig_stats")
+    minlength = sample_config.get("minCtgLength", 1000)
+    sequence = sample_config["reference"]
+    genomesize = sample_config["genomeSize"]
+
+    ctg = re.sub("scf.fasta$", "ctg.fasta", sequence)
+    scf = re.sub("ctg.fasta$", "scf.fasta", sequence)
+
+    def asm_stats(sequence):
+        stats = OrderedDict()
+        stats["assembly type"] = "" 
+        stats["# sequences"] = 0 
+        stats["assembly length"] = 0
+        stats["trim shorter than(bp)"] = minlength
+        stats["# trimmed sequences"] = 0 
+        stats["trimmed assembly length"] = 0
+        stats["N50"] = 0 
+        stats["N80"] = 0 
+        stats["NG50"] = 0 
+        stats["NG80"] = 0
+        stats["longest seq"] = 0
+        sequence_lengths = []
+
+        with open(sequence, "r") as seq_file:
+            # Groupby iterator. Should work for fasta of any column width 
+            fai = groupby(seq_file, lambda x: x.startswith(">"))
+            while True:
+                try:
+                    _, header = fai.next()
+                    _, sequence = fai.next()
+                except StopIteration:
+                    break
+                # Collect fasta sequence stats
+                seq_len = sum([len(i.strip()) for i in sequence])
+                sequence_lengths.append(seq_len)
+                stats["# sequences"] += 1
+                stats["assembly length"] += seq_len
+                if seq_len > minlength:
+                    stats["# trimmed sequences"] += 1
+                    stats["trimmed assembly length"] += seq_len
+                if seq_len > stats["longest seq"]:
+                    stats["longest seq"] = seq_len
```

Comment:
also you could do `stats["longest seq"] = max(sequence_lengths)` outside the with ;)